aws/logs_monitoring/lambda_function.py

```diff
@@ -175,9 +175,19 @@ def extract_trace_payload(event):
     try:
         message = event["message"]
         obj = json.loads(event["message"])
-        if not "traces" in obj or not isinstance(obj["traces"], list):
-            return None
-        return {"message": message, "tags": event[DD_CUSTOM_TAGS]}
+
+        obj_has_traces = "traces" in obj
+        traces_is_a_list = isinstance(obj["traces"], list)
+        # check that the log is not containing a trace array unrelated to Datadog
+        trace_id_found = (
+            len(obj["traces"]) > 0
+            and len(obj["traces"][0]) > 0
+            and obj["traces"][0][0]["trace_id"] is not None
```

Comment:
Could even validate the trace_id format/length?

Reply:
According to this : https://docs.datadoghq.com/tracing/faq/span_and_trace_id_format 
It seems that it's gonna be a large number and I'm afraid that a too restricted test would prevent valid traces from being sent. It seems that the agent is only checking that the trace_id is not zero here : https://docs.datadoghq.com/tracing/faq/span_and_trace_id_format/

Maybe I can add a check on the `resource` field which is not allowed to be empty?